src/mock/MockAllowanceTransferWithCustomChainID.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {ERC20} from "../../lib/solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "../../lib/solmate/src/utils/SafeTransferLib.sol";
+import {PermitHash} from ".././libraries/PermitHash.sol";
+import {SignatureVerification} from ".././libraries/SignatureVerification.sol";
+import {MockEIP712WithCustomChainID} from "./MockEIP712WithCustomChainID.sol";
+import {IAllowanceTransfer} from "../../src/interfaces/IAllowanceTransfer.sol";
+import {SignatureExpired, InvalidNonce} from "../PermitErrors.sol";
+import {Allowance} from ".././libraries/Allowance.sol";
+
+contract MockAllowanceTransferWithCustomChainID is IAllowanceTransfer, MockEIP712WithCustomChainID {
```

Comment:
Why do you need this contract at all? From what I see it's just a copy-paste from the `AllowanceTransfer`. It is also not used anywhere, except `MockPermit2WithCustomChainID`, which is used in one test, where no from copy-pasted functions are used.

Reply:
Sure, I think we can directly use EIP712WithCustomChainID in tests. This simplifies the understanding of the test cases.